components/hab/src/cli.rs

```diff
@@ -264,6 +264,46 @@ pub fn get() -> App<'static, 'static> {
                     (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
                 )
             )
+            (@subcommand secret =>
+                (about: "Commands related to secret management")
+                (@setting ArgRequiredElseHelp)
+                (@subcommand upload => 
+                    (about: "Create and upload a secret for your origin")
+                    (@arg KEY_NAME: +required +takes_value
+                        "The name of the variable key to be injected into the studio.")
+                    (@arg SECRET: +required +takes_value
```

Comment:
Potential future idea (which might be terrible): I wonder what this would look like if you optionally allowed for `stdin` of the secret value so that you could pipe it in like `command-that-gets-secret | hab origin secret AWS_SECRET`. That's just freethinking though…